### ISSUE TITLE
Fails to build: "error: reference to 'filesystem' is ambiguous"

### ISSUE THREAD

**Issue by nunotexbsd** [amsynth/amsynth#233]
Hello,

I'm maintaining amsynth FreeBSD [port](https://www.freshports.org/audio/amsynth/).

Builds are fine on FreeBSD 14.0, 13.3 and 13.2 releases.
I'm getting build error on 15.0 that is main version under development.

I've published logs on our mailings and I've got this response that I'd like that upstream take a look and analyze:

> It looks like the program defines its own "filesystem" class, and also
> has "using namespace std;".
> 
> Usually the easiest fix is to use "::filesystem" for the call sites that
> want to use the program's own definition.
> 
> Alternatively, rename the 'local' definition to something else, like
> "my_filesystem". 

Thanks,
eduardo

### Expected behavior

Build ok

### Actual behavior

```
mv -f src/GUI/.deps/amsynth_dssi_gtk-bitmap_popup.Tpo src/GUI/.deps/amsynth_dssi_gtk-bitmap_popup.Po
cc -DHAVE_CONFIG_H -I.  -DDATADIR='"/usr/local/share"' -DPKGDATADIR='"/usr/local/share/amsynth"' -DPACKAGE_LOCALEDIR=\""/usr/local/share/locale"\" -isystem /usr/local/include -fPIC -fvisibility=hidden -I"./vendor
" -DGDK_DISABLE_DEPRECATED -DGSEAL_ENABLE -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -I/usr/local/include -I/usr/local/include/gtk-2.0 -I/usr/local/include/pango-1.0 -I/usr/local/include/atk-1.0 -I/us
r/local/include/gdk-pixbuf-2.0 -I/usr/local/include -pthread -I/usr/local/include/fribidi -I/usr/local/include/cairo -I/usr/local/include/pixman-1 -D_THREAD_SAFE -I/usr/local/include/harfbuzz -I/usr/local/include
/freetype2 -I/usr/local/include/libpng16 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -isystem /usr/local/include -Wno-declaration-after-statement -Wno-strict-prototypes -O
2 -pipe  -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -MT src/GUI/amsynth_dssi_gtk-editor_pane.o -MD -MP -MF src/GUI/.deps/amsynth_dssi_gtk-editor_pane.Tpo -c -o src/GUI/amsynth_dssi
_gtk-editor_pane.o `test -f 'src/GUI/editor_pane.c' || echo './'`src/GUI/editor_pane.c
src/Configuration.cpp:35:20: error: reference to 'filesystem' is ambiguous
   35 |         amsynthrc_fname = filesystem::get().config;
      |                           ^
src/filesystem.h:27:7: note: candidate found by name lookup is 'filesystem'
   27 | class filesystem
      |       ^
/usr/include/c++/v1/__chrono/file_clock.h:49:1: note: candidate found by name lookup is 'std::filesystem'
   49 | _LIBCPP_BEGIN_NAMESPACE_FILESYSTEM
      | ^
/usr/include/c++/v1/__config:892:80: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_FILESYSTEM'
  892 |                                              inline namespace __fs { namespace filesystem {
      |                                                                                ^
src/Configuration.cpp:60:22: error: reference to 'filesystem' is ambiguous
   60 |         current_bank_file = filesystem::get().default_bank;
      |                             ^
src/filesystem.h:27:7: note: candidate found by name lookup is 'filesystem'
   27 | class filesystem
      |       ^
/usr/include/c++/v1/__chrono/file_clock.h:49:1: note: candidate found by name lookup is 'std::filesystem'
   49 | _LIBCPP_BEGIN_NAMESPACE_FILESYSTEM
      | ^
/usr/include/c++/v1/__config:892:80: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_FILESYSTEM'
  892 |                                              inline namespace __fs { namespace filesystem {
      |                                                                                ^
src/PresetController.cpp:474:9: error: reference to 'filesystem' is ambiguous
  474 |         return filesystem::get().user_banks;
      |                ^
src/filesystem.h:27:7: note: candidate found by name lookup is 'filesystem'
   27 | class filesystem
      |       ^
/usr/include/c++/v1/__chrono/file_clock.h:49:1: note: candidate found by name lookup is 'std::filesystem'
   49 | _LIBCPP_BEGIN_NAMESPACE_FILESYSTEM
      | ^
/usr/include/c++/v1/__config:892:80: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_FILESYSTEM'
  892 |                                              inline namespace __fs { namespace filesystem {
      |                                                                                ^
2 errors generated.
```
Full log: 
[amsynth-1.13.3.log](https://github.com/amsynth/amsynth/files/15174937/amsynth-1.13.3.log)

### Information

FreeBSD 15.0 (1500018) amd64
clang 18.1.4

**Comment by nickdowell** [comment]
Hi Eduardo,

I've pushed a candidate fix to a new branch `hotfix/1.13.4` which removes the `using namespace std` statements.

Any chance you could verify whether it fixes your build issue?

Thanks,
Nick

**Comment by nunotexbsd** [comment]
Hello!

I can confirm that patch fixes 15 (main) builds and runtest is OK.

While this issue affected only 15 main (devel), I'm using fix globally on not affected releases versions too.
No warnings on logs.

https://cgit.freebsd.org/ports/commit/?id=afa68fcf2597d688339477a1ad5e8032c9cddc87

Thanks!
